## Rubric

Instructions: DELETE this cell before you submit via a `git push` to your repo before deadline. This cell is for your reference only and is not needed in your report. 

Scoring: Out of 10 points

- Each Developing  => -2 pts
- Each Unsatisfactory/Missing => -4 pts
  - until the score is 

If students address the detailed feedback in a future checkpoint they will earn these points back


|                  | Unsatisfactory                                                                                                                                                                                                    | Developing                                                                                                                                                                                              | Proficient                                     | Excellent                                                                                                                              |
|------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|
| Data relevance   | Did not have data relevant to their question. Or the datasets don't work together because there is no way to line them up against each other. If there are multiple datasets, most of them have this trouble | Data was only tangentially relevant to the question or a bad proxy for the question. If there are multiple datasets, some of them may be irrelevant or can't be easily combined.                       | All data sources are relevant to the question. | Multiple data sources for each aspect of the project. It's clear how the data supports the needs of the project.                         |
| Data description | Dataset or its cleaning procedures are not described. If there are multiple datasets, most have this trouble                                                                                              | Data was not fully described. If there are multiple datasets, some of them are not fully described                                                                                                      | Data was fully described                       | The details of the data descriptions and perhaps some very basic EDA also make it clear how the data supports the needs of the project. |
| Data wrangling   | Did not obtain data. They did not clean/tidy the data they obtained.  If there are multiple datasets, most have this trouble                                                                                 | Data was partially cleaned or tidied. Perhaps you struggled to verify that the data was clean because they did not present it well. If there are multiple datasets, some have this trouble | The data is cleaned and tidied.                | The data is spotless and they used tools to visualize the data cleanliness and you were convinced at first glance                      |


# COGS 108 - Data Checkpoint

## Authors

Instructions: REPLACE the contents of this cell with your team list and their contributions. Note that this will change over the course of the checkpoints

This is a modified [CRediT taxonomy of contributions](https://credit.niso.org). For each group member please list how they contributed to this project using these terms:
> Analysis, Background research, Conceptualization, Data curation, Experimental investigation, Methodology, Project administration, Software, Visualization, Writing – original draft, Writing – review & editing

Example team list and credits:
- Alice Anderson: Conceptualization, Data curation, Methodology, Writing - original draft
- Bob Barker:  Analysis, Software, Visualization
- Charlie Chang: Project administration, Software, Writing - review & editing
- Dani Delgado: Analysis, Background research, Visualization, Writing - original draft

## Research Question

Among adults aged 18 years and older in the United States, how is average weekly intentional exercise (measured in minutes per week) associated with prevalence of obesity, diabetes, and self-reported mental health outcomes?


## Background and Prior Work

Health outcomes vary across the United States and are influenced by a combination of biological and lifestyle factors. Conditions like diabetes, obesity, and mental health are not attributed to a single cause. Factors like genetics, medical conditions, access to healthcare and the daily routine of an individual all play a role. One thing we can say about is that physical activity is one of the factors that can easily improve such conditions when time is dedicated to it weekly, and our goal is to examine how physical activity is associated with adult obesity prevalence, diabetes prevalence, and mental health across the U.S.
Physical activity plays an important role in maintaining both physical and mental health and it is recognized as an important health factor. Physical activity includes everyday movements such as walking, running, stretching, or even having a structured workout. According to the World Health Organization (WHO), individuals should engage in at least 150 minutes of moderate to intense physical activity per week to support health outcomes <a name="cite_ref-1"></a><a href="#cite_note-1">1</a>. Despite the recommendations from this organization, the amount of time or levels of physical activity varies across the individuals in the U.S. due to the differences they have in their work schedules, access to resources, and personal life.

Previous research has shown a beneficial relationship between higher levels of physical activity and lower prevalence of obesity and diabetes. The Center for Disease Control and Prevention (CDC) has reported that obesity and diabetes prevalence have steadily increased across the U.S. over the past decades<a name="cite_ref-2"></a><a href="#cite_note-2"> 2</a>. We noticed that these trends are influenced by a variety of factors which include diet quality, limited physical activity, metabolic conditions like thyroid conditions, or even the individual's metabolism. However, in this project, we focus on physical activity as the key variable and examine how it is associated with obesity and diabetes.

Moving along with physical health, there is also prior work that examined the relationship between physical activity and mental health outcomes. Research from the Harvard T.H. Chan School of Public Health found that even modest amounts of physical activity, such as 15 minutes of exercise per day or even one hour of walking, are associated with reduced risk of major depression and improved overall mental health outcomes<a name="cite_ref-3"></a><a href="#cite_note-3"> 3</a>. This supports the inclusion of mental health as an outcome variable in our project, along with obesity and diabetes when we examine how physical activity varies across the U.S.

Our group has noticed that the existing research focuses more on the mental and health benefits for an individual. Our goal is to show how physical activity helps in a broader geographic level which is in the U.S. The differences in physical activity rates across the states may help explain the observed variation in obesity prevalence, diabetes prevalence and mental health indicators. This project aims to identify patterns that may help inform public health strategies and highlight the disparities in health and mental outcomes related to physical activity.

1. <a name="cite_note-1"></a> [^](#cite_ref-1) World Health Organization. (30 Aug 2020). 
<i>Physical activity</i>. https://www.who.int/initiatives/behealthy/physical-activity

3. <a name="cite_note-2"></a> [^](#cite_ref-2) Centers for Disease Control and Prevention. (28 Jan 2026). 
<i>National Diabetes Statistics Report</i>. https://gis.cdc.gov/grasp/diabetes/diabetesatlas-statsreport.html


5. <a name="cite_note-3"></a> [^](#cite_ref-3) Harvard Health Publishing. (1 May 2019). 
<i>More evidence that exercise can boost mood</i>. https://www.health.harvard.edu/mind-and-mood/more-evidence-that-exercise-can-boost-mood




## Hypothesis


We predict a strong negative correlation between minutes of intentional exercise and obesity rates and diabetes diagnosis rates among U.S. adults. We also predict that higher minutes of exercise will be positively correlated with improved self-reported mental health outcomes. Additionally, we further hypothesize that these relationships will vary by factors such as age and gender. Specifically, we predict that men between the ages 18 and 30 have higher weekly intentional exercise minutes per week and correspondingly lower obesity and diabetes rates, which will be positively correlated with self-reported mental health outcomes. In contrast, we also predict that women aged 40 and older will have lower amounts of weekly intentional exercise minutes and higher obesity and diabetes rates, which will be negatively correlated with self-reported mental health outcomes.

## Data

### Data overview

Instructions: REPLACE the contents of this cell with descriptions of your actual datasets.

For each dataset include the following information
- Dataset #1
  - Dataset Name:
  - Link to the dataset:
  - Number of observations:
  - Number of variables:
  - Description of the variables most relevant to this project
  - Descriptions of any shortcomings this dataset has with repsect to the project
- Dataset #2 (if you have more than one!)
  - same as above
- etc

Each dataset deserves either a set of bullet points as above or a few sentences if you prefer that method.

If you plan to use multiple datasets, add a few sentences about how you plan to combine these datasets.

In [1]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

#%load_ext autoreload
#%autoreload 2

In [2]:
# Setup code -- this only needs to be run once after cloning the repo!
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
datafiles = [
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/airline-safety/airline-safety.csv', 'filename':'airline-safety.csv'},
    { 'url': 'https://raw.githubusercontent.com/fivethirtyeight/data/refs/heads/master/bad-drivers/bad-drivers.csv', 'filename':'bad-drivers.csv'}
]

get_data.get_raw(datafiles,destination_directory='data/00-raw/')

Overall Download Progress:  50%|█████     | 1/2 [00:00<00:00,  8.36it/s]   

Successfully downloaded: airline-safety.csv



Overall Download Progress: 100%|██████████| 2/2 [00:00<00:00,  7.80it/s]

Successfully downloaded: bad-drivers.csv


### Dataset #1 

Instructions: 
1. Change the header from Dataset #1 to something more descriptive of the dataset
2. Write a few paragraphs about this dataset. Make sure to cover
   1. Describe the important metrics, what units they are in, and giv some sense of what they mean.  For example "Fasting blood glucose in units of mg glucose per deciliter of blood.  Normal values for healthy individuals range from 70 to 100 mg/dL.  Values 100-125 are prediabetic and values >125mg/dL indicate diabetes. Values <70 indicate hypoglycemia. Fasting idicates the patient hasn't eaten in the last 8 hours.  If blood glucose is >250 or <50 at any time (regardless of the time of last meal) the patient's life may be in immediate danger"
   2. If there are any major concerns with the dataset, describe them. For example "Dataset is composed of people who are serious enough about eating healthy that they voluntarily downloaded an app dedicated to tracking their eating patterns. This sample is likely biased because of that self-selection. These people own smartphones and may be healthier and may have more disposable income than the average person.  Those who voluntarily log conscientiously and for long amounts of time are also likely even more interested in health than those who download the app and only log a bit before getting tired of it"
3. Use the cell below to 
    1. load the dataset 
    2. make the dataset tidy or demonstrate that it was already tidy
    3. demonstrate the size of the dataset
    4. find out how much data is missing, where its missing, and if its missing at random or seems to have any systematic relationships in its missingness
    5. find and flag any outliers or suspicious entries
    6. clean the data or demonstrate that it was already clean.  You may choose how to deal with missingness (dropna of fillna... how='any' or 'all') and you should justify your choice in some way
    7. You will load raw data from `data/00-raw/`, you will (optionally) write intermediate stages of your work to `data/01-interim` and you will write the final fully wrangled version of your data to `data/02-processed`
4. Optionally you can also show some summary statistics for variables that you think are important to the project
5. Feel free to add more cells here if that's helpful for you


In [3]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE


### Physical Activity and its connection with Mental Health outcomes for individuals in the U.S.
Within this dataset it contains each individual level information such as, lifestyle behaviors, physical activity habits, and self reports on their own mental health. The key physical metrics in this dataset are Daily_Steps, which is the average number of steps per day, Exercise_duaration which is average duration per session in minutes, and Exercise_frequency which is days per week the person exercises (0-7). By using the above variables, the total number of minutes participants exercised during a week may be determined, and total exercise time can be compared with recommended exercise levels by public health organizations for adults, such as the health department recommends at least 150 minutes per week of moderate-intensity aerobic activity for adults. The life style variables include Sleep_Hours which is the average sleep per night in hours, Screen_Time_Hours which is the screen time per day (non-work screen use), Diet_Quality which is a self rated diet quality (1=poor, 5=excellent), and finally Social_Interaction which is the number of meaningful social interactions per week. Metrics like those listed above are important indicators of your overall health, and how healthy you are physically and mentally. Also to help provide context when interpreting exercise effects.

Mental health within the dataset is measured using a self-reported scale from 1-10. The four self-reported mental health measurements are: Stress_Level, Anxiety_Level, Depression_Level, and Happiness_Level, where at 10 represents the highest intensity of the condition. The dataset also includes a calculated Mental_Health_Score, defined as (Happiness × 2) − ((Stress + Anxiety + Depression) / 3). This composite score is an attempt to capture overall well-being by giving more weight to happiness and subtracting the mean of negative emotional experiences. Positive values indicate better mental health, and negative values indicate poorer well-being.But these ratings are still subjective, and are not clinically validated or diagnosed. So the data should just be interpreted as just simple indicators rather than a proven medical diagnosis. 

In the dataset there are several concerns that we as a group have noticed. Within the dataset the sample size is relatively pretty small (300 participants), which limits generalizability. This also causes a concern because the conclusions are meant to represent the broader U.S. adult population. There's uncertainty regarding how participants were chosen ( randomly chosen,or  through volunteer participation), and therefore, there is a possibility of sampling bias. Participants willing to share their health and fitness data may be more inclined to stay healthy, may have greater access to technology, and different socioeconomic status than people in the general population.Moreover, a considerable percentage of the variables obtained from participants are self-reported, which presents the potential for recall bias and social desirability bias. Additionally, given that the data set author developed their own scale for measuring mental health outcomes without validation, there are valid concerns that the subsequent data collected as a result of this measure may lack reliable measurement. Due to these issues, it would be good to use findings generated from this dataset in an exploratory correlational manner as opposed to being used definitively or as demonstrating causal.


In [3]:
## YOUR CODE TO LOAD/CLEAN/TIDY/WRANGLE THE DATA GOES HERE
import pandas as pd
activity_mental = pd.read_csv('data/00-raw/mental_health_physical_activity.csv')
activity_mental.head()

ID  Age  Gender            Occupation  Sleep_Hours  Daily_Steps  \
0   1   58    Male               Student          7.7         5012   
1   2   23    Male  Working Professional          6.5         1434   
2   3   62  Female            Unemployed          5.4         4527   
3   4   25  Female               Student          6.8        14588   
4   5   35  Female            Unemployed          4.8         6820   

   Exercise_Frequency  Exercise_Duration Exercise_Type  Screen_Time_Hours  \
0                   3               12.6        Cardio                7.1   
1                   3               64.4           NaN                5.9   
2                   5                9.2        Sports                1.9   
3                   5               52.0           NaN                1.6   
4                   3               60.3           NaN                9.4   

   Diet_Quality  Social_Interaction  Stress_Level  Anxiety_Level  \
0             5                   2            10              7   
1             2                  14            10              5   
2             3                  19             5              1   
3             2                   9             2              4   
4             1                  19             3              9   

   Depression_Level  Happiness_Level  Mental_Health_Score  Notes  
0                 1                1                 -4.0    NaN  
1                 1                3                  0.7    NaN  
2                 8                9                 13.3    NaN  
3                 2                7                 11.3    NaN  
4                 4                3                  0.7    NaN

In [4]:
activity_mental.columns

Index(['ID', 'Age', 'Gender', 'Occupation', 'Sleep_Hours', 'Daily_Steps',
       'Exercise_Frequency', 'Exercise_Duration', 'Exercise_Type',
       'Screen_Time_Hours', 'Diet_Quality', 'Social_Interaction',
       'Stress_Level', 'Anxiety_Level', 'Depression_Level', 'Happiness_Level',
       'Mental_Health_Score', 'Notes'],
      dtype='object')

In [5]:
print(activity_mental['ID'].duplicated().sum())
activity_mental.shape

0


(300, 18)

After checking the columns and the shape of the dataset, we can see that there are no duplicate ID's or merged columns that would make our dataset untidy. Since the duplicated check returned 0, we can safely confirm this data set is already tidy. Now we proceed to check how many missing values are on this dataset.

In [6]:
missing_values_am = activity_mental.isna().sum()
missing_values_am


ID                       0
Age                      0
Gender                   0
Occupation               0
Sleep_Hours              0
Daily_Steps              0
Exercise_Frequency       0
Exercise_Duration        0
Exercise_Type           48
Screen_Time_Hours        0
Diet_Quality             0
Social_Interaction       0
Stress_Level             0
Anxiety_Level            0
Depression_Level         0
Happiness_Level          0
Mental_Health_Score      0
Notes                  300
dtype: int64

After checking the missing values in the dataset, we found that 48 individuals did not specify their Exercise_Type. We can safely remove these rows since Exercise_Type is not our primary variable of interest. We may revisit this column later to determine whether a certain Exercise_Type is associated with more positive mental health outcomes. The "Notes" column is irrelevant to this project and will be dropped entirely since all 300 values are missing.

In [7]:
clean_am = activity_mental.dropna(subset = 'Exercise_Type').drop(columns = 'Notes')
clean_am

ID  Age  Gender            Occupation  Sleep_Hours  Daily_Steps  \
0      1   58    Male               Student          7.7         5012   
2      3   62  Female            Unemployed          5.4         4527   
5      6   47  Female            Unemployed          8.9        11485   
7      8   65   Other               Retired          8.5         7543   
8      9   56    Male               Retired          5.9         8668   
..   ...  ...     ...                   ...          ...          ...   
292  293   26  Female            Unemployed          8.2         9975   
293  294   63   Other  Working Professional          6.5        12375   
295  296   62  Female  Working Professional          7.8         1529   
297  298   46    Male            Unemployed          7.8         9955   
298  299   42    Male               Retired          5.4         5329   

     Exercise_Frequency  Exercise_Duration Exercise_Type  Screen_Time_Hours  \
0                     3               12.6        Cardio                7.1   
2                     5                9.2        Sports                1.9   
5                     3               61.6        Cardio                3.1   
7                     5               19.7      Strength                5.6   
8                     4               87.4        Cardio                7.1   
..                  ...                ...           ...                ...   
292                   5               64.1         Mixed                6.7   
293                   5               63.9          Yoga                6.1   
295                   5               73.3        Cardio                6.5   
297                   3               69.9      Strength                3.6   
298                   2               14.5          Yoga                1.1   

     Diet_Quality  Social_Interaction  Stress_Level  Anxiety_Level  \
0               5                   2            10              7   
2               3                  19             5              1   
5               1                  10             7              5   
7               1                   1             2              3   
8               1                  17             5              6   
..            ...                 ...           ...            ...   
292             2                   6            10              5   
293             5                  18             7              9   
295             5                   7             9              7   
297             2                   2             3              3   
298             4                   4             1              3   

     Depression_Level  Happiness_Level  Mental_Health_Score  
0                   1                1                 -4.0  
2                   8                9                 13.3  
5                   2                4                  3.3  
7                   3                7                 11.3  
8                   2                5                  5.7  
..                ...              ...                  ...  
292                 6                3                 -1.0  
293                 5                7                  7.0  
295                 4                9                 11.3  
297                 9                2                 -1.0  
298                 6               10                 16.7  

[252 rows x 17 columns]

We can now see a clean and tidy dataset that showcases the relationship between the amount of physical activity done weekly and the mental health outcome score. The drop() function was used for the 'Notes' column since all values were missing. For the column 'Exercise_Type' there were 48 missing values, so it is safe to use the dropna() function which removes the complete row since we may need to check whether a certain exercise type improves mental health outcomes. Even though it is not our primary variable of interest, it is safe to revisit this later in the project.

Let us now inspect the "Mental_Health_Score" column

In [8]:
mh_score = clean_am['Mental_Health_Score']
mh_score.describe()

count    252.000000
mean       5.930159
std        6.062145
min       -6.000000
25%        0.600000
50%        6.150000
75%       11.300000
max       17.700000
Name: Mental_Health_Score, dtype: float64

The "Mental_Health_Score" values ranges from -6.0 to 17.7 with a mean of 5.93. We flag this as a concern because these values have no clear clinical representation. This dataset does not have an established scale to determine what are the good or bad scores look like. This score is calculated by a formula created by the author rather than a validated mental health instrument. It is safe to use the self-reported values columns instead. Therefore, we can proceed by dropping the table and only using the self-reported values.

In [9]:
clean_am_sr = clean_am.drop(columns = 'Mental_Health_Score')
clean_am_sr.head()

ID  Age  Gender  Occupation  Sleep_Hours  Daily_Steps  Exercise_Frequency  \
0   1   58    Male     Student          7.7         5012                   3   
2   3   62  Female  Unemployed          5.4         4527                   5   
5   6   47  Female  Unemployed          8.9        11485                   3   
7   8   65   Other     Retired          8.5         7543                   5   
8   9   56    Male     Retired          5.9         8668                   4   

   Exercise_Duration Exercise_Type  Screen_Time_Hours  Diet_Quality  \
0               12.6        Cardio                7.1             5   
2                9.2        Sports                1.9             3   
5               61.6        Cardio                3.1             1   
7               19.7      Strength                5.6             1   
8               87.4        Cardio                7.1             1   

   Social_Interaction  Stress_Level  Anxiety_Level  Depression_Level  \
0                   2            10              7                 1   
2                  19             5              1                 8   
5                  10             7              5                 2   
7                   1             2              3                 3   
8                  17             5              6                 2   

   Happiness_Level  
0                1  
2                9  
5                4  
7                7  
8                5

We know have a clean and tidy table where there are no merging columns and missing values. We have the sufficient information to get started with out project. We can proceed to show the statistics of some of the important variables we will be using to conduct or hypothesis which are the following: 

In [10]:
clean_am_sr[
['Age','Gender',
'Exercise_Frequency','Exercise_Duration',
'Stress_Level','Anxiety_Level',
'Depression_Level','Happiness_Level']].describe()

Age  Exercise_Frequency  Exercise_Duration  Stress_Level  \
count  252.000000          252.000000         252.000000    252.000000   
mean    42.595238            3.619048          40.581746      5.785714   
std     14.371662            2.187823          28.374128      2.880369   
min     18.000000            0.000000           0.000000      1.000000   
25%     30.000000            2.000000          13.550000      3.000000   
50%     42.000000            4.000000          40.600000      6.000000   
75%     56.250000            5.000000          64.950000      8.000000   
max     65.000000            7.000000          89.600000     10.000000   

       Anxiety_Level  Depression_Level  Happiness_Level  
count     252.000000        252.000000       252.000000  
mean        5.428571          5.226190         5.706349  
std         2.782783          2.902195         2.999515  
min         1.000000          1.000000         1.000000  
25%         3.000000          3.000000         3.000000  
50%         5.000000          5.000000         6.000000  
75%         8.000000          8.000000         8.000000  
max        10.000000         10.000000        10.000000

In [11]:
clean_am_sr.to_csv('data/02-processed/mental_health_physical_activity_clean.csv', index=False)

## Ethics

### A. Data Collection
- [x] **A.1 Informed consent**: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data uses to which they consent?

We will not use any human-subject interaction or individual-level data for this project, thus no informed consent is required. All analysis is performed at aggregated (U.S. state) level using publicly available, non-interactive data sources.

- [x] **A.2 Collection bias**: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?

We have acknowledged that the underlying datasets may reflect collection and survey biases (e.g., nonresponse, sampling bias). We will document known limitations of each dataset and avoid over-interpreting state-level aggregates where collection bias may distort the results.

- [x] **A.3 Limit PII exposure**: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?

Datasets used are at the state-aggregate level and contain no personally identifiable information. We will not attempt to merge data in ways that could reconstruct individual identities.

- [x] **A.4 Downstream bias mitigation**: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

Some data sources include demographic variables that allow subgroup comparisons. While our primary analysis is state-level, we recognize that aggregation can conceal disparities. We will note these limitations and, where possible, examine demographic breakdowns to identify patterns that aggregate measures might hide.

### B. Data Storage
- [x] **B.1 Data security**: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?

All datasets are publicly available and non-sensitive, but we will store copies on password-protected personal and institutional systems and follow standard security practices to reduce risk of unauthorized access.

- [x] **B.2 Right to be forgotten**: Do we have a mechanism through which an individual can request their personal information be removed?

No individual-level personal data are collected or stored. Because no PII is used, a right-to-be-forgotten mechanism is not required; nevertheless, we will delete any derived files if requested and as described in our retention plan.

- [x] **B.3 Data retention plan**: Is there a schedule or plan to delete the data after it is no longer needed?

We do not plan to retain the project beyond the course. All project files and data will be deleted after the course concludes, unless the team agrees to preserve a cleaned, anonymized copy for future educational purposes and documents that decision.

### C. Analysis
- [x] **C.1 Missing perspectives**: Have we sought to address blindspots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?

We acknowledge that state-level analyses miss local and marginalized perspectives. We will explicitly call out these blindspots in our write-up and interpret findings conservatively, avoiding claims about individual-level causation.

- [x] **C.2 Dataset bias**: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?

We will examine missingness patterns, consistency across states, and known survey biases (e.g., recall bias, social desirability) and document these issues. Where feasible, we will avoid analyses that amplify biased signals and will be transparent about remaining biases.

- [x] **C.3 Honest representation**: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?

All visualizations and statistics will be produced to accurately reflect the data and the uncertainty in estimates; we will include appropriate labels, scales, and caveats so readers are not misled.

- [x] **C.4 Privacy in analysis**: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?

No PII will be used or displayed in any analysis or figures.

- [x] **C.5 Auditability**: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?

We will document all data cleaning, merging, and analysis steps and provide code and notes to enable reproducibility and auditing of our results.

### D. Modeling
- [x] **D.1 Proxy discrimination**: Have we ensured that the model does not rely on variables or proxies for variables that are unfairly discriminatory?

We considered whether physical activity and health outcomes could act as proxies for socioeconomic status or access to care and will interpret associations cautiously, avoiding language that blames individuals or states for structural disparities.

- [x] **D.2 Fairness across groups**: Have we tested model results for fairness with respect to different affected groups (e.g., tested for disparate error rates)?

Because our analysis is at the state-aggregate level, fairness testing across individual demographic groups is limited. Where subgroup data are available, we will report descriptive comparisons and caveat the limitations of such comparisons.

- [ ] **D.3 Metric selection**: Have we considered the effects of optimizing for our defined metrics and considered additional metrics?

We have not defined modeling metrics for a deployed decision system; this item is not applicable for our current exploratory/state-level analyses.

- [ ] **D.4 Explainability**: Can we explain in understandable terms a decision the model made in cases where a justification is needed?

Not applicable for a deployed decision model; we will, however, ensure that any statistical models used are interpretable and that we explain key relationships in plain language.

- [x] **D.5 Communicate limitations**: Have we communicated the shortcomings, limitations, and biases of the model to relevant stakeholders in ways that can be generally understood?

We will explicitly communicate limitations of our analysis and potential biases in the report and presentation to avoid overgeneralization or misuse.

### E. Deployment
- [x] **E.1 Monitoring and evaluation**: Do we have a plan to monitor the model and its impacts after it is deployed?

This project is exploratory and not intended for deployment in decision-making. If outputs were to be used beyond this course, additional monitoring and validation procedures would be required and are outside the scope of the current work.

- [x] **E.2 Redress**: Have we discussed with our organization a plan for response if users are harmed by the results?

No direct users are affected by this class project; nonetheless, if interpretations or visualizations cause misinterpretation, we will correct and clarify our findings promptly in our documentation and public materials.

- [ ] **E.3 Roll back**: Is there a way to turn off or roll back the model in production if necessary?

Not applicable because we are not deploying a production model.

- [x] **E.4 Unintended use**: Have we taken steps to identify and prevent unintended uses and abuse of the model and do we have a plan to monitor these once the model is deployed?

We recognize results could be misused to place individual accountability for structural health outcomes. To reduce misuse, we will provide clear context, document limitations, and emphasize structural determinants of health in our report.

## Team Expectations 


Ava, Bryan, Celio, Austin, Mary


Read the [COGS108 Team Policies](https://github.com/COGS108/Projects/blob/master/COGS108_TeamPolicies.md) individually. By including each member’s name above and adding their name to the submission, each member indicates they have read the COGS108 Team Policies, accept these team expectations, and intend to fulfill them.

Team Expectation 1 — Communication
- We will communicate via iMessage and FaceTime. We will communicate respectfully and promptly.
- We expect to meet one to two times per week on FaceTime to review project components and will text throughout the week to check in and coordinate.
- If a member cannot attend a FaceTime meeting, they should review meeting notes and follow up via text as soon as possible.

Team Expectation 2 — Tone and Conflict
- Our tone will be polite, professional, and honest. We prioritize respect first and clear, direct communication second.
- If conflicts arise, we will handle them professionally and directly, clarifying expectations to avoid misunderstandings.

Team Expectation 3 — Decision Making
- Decisions will be made by majority vote (at least 3 of 5 members) to ensure fairness and timely progress.
- Exception: selecting the project idea requires unanimous agreement from all five members.
- If a decision must be made quickly, the majority of members present at that time may decide. If an absent member does not respond promptly via text, we will proceed without them.

Team Expectation 4 — Roles and Learning
- Members will naturally focus on tasks that match their strengths, but we will strive to collaborate so everyone learns all parts of the project.
- We will share knowledge and pair on tasks when feasible to distribute learning.

Team Expectation 5 — Timeline and Documentation
- We will maintain and update the project timeline to stay organized.
- The timeline will be reviewed and adjusted during our weekly FaceTime meetings.
- We will document meetings and decisions so responsibilities and progress are clear.

Team Expectation 6 — Handling Struggles and Deadlines
- If a member struggles to complete their assigned work, they should notify the group as soon as possible via text.
- When time allows, teammates will help. If we face a tight deadline, we will reassign the task to ensure the project stays on schedule.
- Reassigning work is a last-resort solution to avoid falling behind; we aim to distribute work fairly and support members early.

Team Expectation 7 — Accessibility of Expectations
- We have pasted these expectations into our group text chat and shared them as a file so every member can refer to them throughout the quarter.

## Project Timeline Proposal

## Project Timeline Proposal

Weekly meetings: FaceTime meetings every Tuesday at 8:00 PM (unless otherwise noted). Ad-hoc communications via group iMessage for scheduling, quick questions, and file sharing. Major milestones and meeting notes will be recorded in the repo (meetings/ or notes/ folder) after each meeting.

| Meeting Date | Meeting Time | Completed Before Meeting | Discussed at Meeting |
|---|---:|---|---|
| 1/26 | 6:00 PM | Looked for CSV files related to fitness and health issues; ensured everyone had the repo on GitHub | Project theme; research question; CSV file availability; repo access |
| 2/03 | 8:00 PM | Formed 3 possible research questions; found CSV files relating to our topic; reviewed two past COGS108 projects | Project proposal scope; split proposal assignments; availability and future meetings |
| 2/04 | 8:00 PM | Completed project proposal; formalized research question; background research | Submission logistics; GitHub branches and pull requests; next steps for data search |
| 2/10 | 8:00 PM | Import & begin wrangling data | Status update; review/edit wrangling; plan for upcoming data checkpoint |
| 2/17 | 8:00 PM | Data checkpoint prep | Data checkpoint; begin detailed data analysis; assess state of the project.|
| 2/24 | 8:00 PM | Complete primary data analysis tasks and submit checkpoint deliverables | Finalize data analysis; plan timeline for finals work and presentation availability |
| 3/10 | 8:00 PM | Draft conclusion; prepare analysis results | Plan final project video; ensure all code/notebooks run; finalize remaining items |
| 3/17 | 8:00 PM | N/A | Turn in final project; complete group evaluation forms |

Planned milestone summary
- Week-to-week rhythm: research/wrangling during weekdays, discuss progress and blockers at Tuesday 8 PM meetings, and assign concrete tasks for the following week.
- Checkpoints:
  - Data checkpoint: 2/17 (in-meeting review of cleaned and merged datasets)
  - Analysis checkpoint: 2/24 (key analysis steps completed; draft figures)
  - Final draft & video planning: 3/10
  - Final submission & evaluations: 3/17
- Responsibilities: we will assign owners for data wrangling, EDA, modeling, visualization, write-up, and the final video; owners will update status in the repo prior to each meeting.

Special resources / training needed
- No special external tools beyond Python (pandas, numpy, matplotlib/seaborn, scikit-learn/statsmodels) and Jupyter notebooks are required. If we use wearable/device study data requiring special access, one member will handle the data-access request process and document required approvals.

Communication & conflict resolution
- Primary channels: iMessage for daily coordination; FaceTime for weekly meetings; GitHub issues and PRs for code review and task tracking.
- If conflicts arise, raise the issue in the group chat and bring it to the next FaceTime meeting; if urgent, schedule a short ad-hoc FaceTime. Decisions use majority vote (3 of 5) except for project topic selection which requires unanimity.